code/game/machinery/deployable.dm

```diff
@@ -254,24 +81,79 @@ for reference:
 		var/obj/item/projectile/proj = mover
 		if(proj.firer && Adjacent(proj.firer))
 			return 1
-		if(prob(20))
+		if(prob(proj_pass_rate))
 			return 1
 		return 0
 	else
 		return 0
 
-/obj/machinery/deployable/barrier/proc/explode()
 
-	visible_message("<span class='danger'>[src] blows apart!</span>")
-	var/turf/Tsec = get_turf(src)
 
-/*	var/obj/item/stack/rods/ =*/
-	new /obj/item/stack/rods(Tsec)
 
-	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
-	s.set_up(3, 1, src)
-	s.start()
 
-	explosion(src.loc,-1,-1,0)
-	if(src)
-		qdel(src)
+/////BARRICADE TYPES///////
+
+/obj/structure/barricade/wooden
+	name = "wooden barricade"
+	desc = "This space is blocked off by a wooden barricade."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "woodenbarricade"

```

Comment:
just replace the welding stuff in attackby() with a call to a being_welded() or something like that, and override being_welded() to just return on the wooden child